### PR TITLE
qtwebengine: link to pulseaudio

### LINKED ISSUE #197728: Virtual devices created in pipewire do not show up as audio inputs in qtwebengine-based browsers
### Describe the bug
I found this bug while trying to create a package for [discord-screenaudio](https://github.com/maltejur/discord-screenaudio) (mentioned in #186957), but it seems all qtwebengine-based browsers like Qutebrowser and Falkon have the same issue.

These browsers are able to pick up my microphone just fine, but any virtual devices created via pipewire do not show up. discord-screenaudio creates a virtual source to pipe desktop audio through, which then is meant to be detected as an audio input inside a qtwebengine window linked to discord.com.

Creating virtual devices manually via `pactl` (following [this guide](https://gitlab.freedesktop.org/pipewire/pipewire/-/wikis/Virtual-Devices)) also doesn't work, as they do not show up in Qutebrowser or Falkon. They do show up in Firefox, Chromium, and the native Discord client, however.

### Steps To Reproduce
1. have pipewire running as the audio server
2. create a virtual audio device using `pactl load-module module-null-sink media.class=Audio/Source/Virtual sink_name=my-virtual-source channel_map=front-left,front-right`
3. go to [mictests.com](https://mictests.com/) on Falkon or Qutebrowser and see that `my-virtual-source` does not show up in the input list.

### Expected behavior
`my-virtual-source` does show up in Falkon or Qutebrowser

### Additional context
One final difference I see is within the Helvum program, where I can see how audio devices are connected. When Firefox, Chromium and Discord use my microphone a connection is made in Helvum, but when Qutebrowser, Falkon, or discord-screenaudio do no connection is visible. To me it seems that qtwebengine-based browsers must not be connecting to pipewire or pipewire-pulse at all, but talking to the audio devices directly somehow.

### Notify maintainers
@matthewbauer (qtwebengine)

### Metadata
Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
> nix-shell -p nix-info --run "nix-info -m"
this path will be fetched (0.06 MiB download, 0.30 MiB unpacked):
  /nix/store/x8mymrkpsmpwyvqssjbwsq851kscf1kw-bash-interactive-5.1-p16-dev
copying path '/nix/store/x8mymrkpsmpwyvqssjbwsq851kscf1kw-bash-interactive-5.1-p16-dev' from 'https://cache.nixos.org'...
 - system: `"x86_64-linux"`
 - host os: `Linux 5.15.74, NixOS, 22.05 (Quokka), 22.05.20221022.3933d8b`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.8.1`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`
```

### PR DESCRIPTION
**Pull request by vlaci**
Nix builds of QTWebEngine doesn't support sound output via Pulse, only through ALSA, unless sandboxing is disabled and LD_LIBRARY_PATH contains a reference to the Pulse libraries.

## Description of changes

By default, Chromium won't link to PulseAudio, but dlopens it in run-time. It is tricky to make this work because of the nontrivial sandbox configuration. Nix builds of Chromium already [link against Pulse directly](https://github.com/NixOS/nixpkgs/blob/5029a01733c7b6bdfe086c39237316566cfe57d2/pkgs/applications/networking/browsers/chromium/common.nix#L397), so it makes sense to copy this behavior in QTWebEngine.

## Things done

I've checked that Qutebrowser has proper PulseAudio support after building with an earlier version of this patch. As building Chromium takes forever, I am yet to test this latest revision.
Possibly resolves #197728

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by vlaci** [comment]
Tested with `qutebrowser` and `qutebrowser-qt6`.

**Comment by K900** [comment]
Can we just combine all the changes into one patch? I don't see why we would want to patch one file and cat another.

**Comment by vlaci** [comment]
> Can we just combine all the changes into one patch? I don't see why we would want to patch one file and cat another.

Those are in two different derivations, so it would be another patch. Honestly I got really confused how QT 5 is packaged, as there is an extra layer of indirection between the module definition and the final [version specific derivation](https://github.com/NixOS/nixpkgs/blob/master/pkgs/development/libraries/qt-5/5.15/default.nix#L280) where the patches reside.

**Comment by K900** [comment]
So the build configs are different enough that we can't use the same patch for both?

**Comment by vlaci** [comment]
> So the build configs are different enough that we can't use the same patch for both?

Exactly: QT 5 is on qmake, while 6 is cmake. The build files have completely different structure.

**Comment by K900** [comment]
Then I'd still prefer using patches, even if different ones, for both, just for consistency (and yes, Qt5 packaging is a mess - it used to be multiversioned, and now it isn't, but the legacy remains).

File: pkgs/development/libraries/qt-5/5.15/default.nix [hunk 1]
```diff
@@ -174,6 +174,8 @@ let
       ./qtwebengine-darwin-no-platform-check.patch
       ./qtwebengine-mac-dont-set-dsymutil-path.patch
       ./qtwebengine-darwin-checks.patch
+    ] ++ lib.optionals stdenv.isLinux [
```
Comment (K900) [inline pkgs/development/libraries/qt-5/5.15/default.nix:177]: Maybe this check could be relaxed to also account for BSDs and such? I'm not sure how many platforms we use Pulseaudio on...

Comment (vlaci) [reply pkgs/development/libraries/qt-5/5.15/default.nix:177]: Actually it is possible to enable it unconditionally everywhere, as it's usage is guarded by the `use_pulseaudio` flag which itself is autodetected during build time on both versions.

```
  if (use_pulseaudio) {
    defines += [ "USE_PULSEAUDIO" ]
    if (!link_pulseaudio) {
      defines += [ "DLOPEN_PULSEAUDIO" ]
    }
  }
```

**Comment by K900** [comment]
Sorry, missed it. LGTM, let's send it.

**Pull request merged**